packages/manager/modules/iam/src/components/advancedPolicySearch/advancedPolicySearch.controller.js

```diff
@@ -1,6 +1,110 @@
 export default class AdvancedPolicySearchController {
   /* @ngInject */
-  constructor(IAMService) {
+  constructor($state, IAMService) {
+    this.$state = $state;
     this.IAMService = IAMService;
+
+    this.model = {
+      selectedIdentities: {},
+      selectedResources: {},
+      selectedActions: {},
+    };
+
+    this.localUsers = { loading: true, search: '', items: [] };
+    this.userGroups = { loading: true, search: '', items: [] };
+
+    this.filterLocalUsersFn = this.filterLocalUsers.bind(this);
+    this.filterUserGroupsFn = this.filterUserGroup.bind(this);
+  }
+
+  $onInit() {
+    const identities = (this.identitiesFilter || []).reduce(
+      (acc, key) => ({ ...acc, [key]: true }),
+      {},
+    );
+    const resources = (this.resourcesFilter || []).reduce(
+      (acc, key) => ({ ...acc, [key]: true }),
+      {},
+    );
+    const actions = (this.actionsFilter || []).reduce(
+      (acc, key) => ({ ...acc, [key]: true }),
+      {},
+    );
+    this.model = {
+      selectedIdentities: identities,
+      selectedResources: resources,
+      selectedActions: actions,
+    };
+
+    this.getLocalUsers().then((localUsers) => {
+      this.localUsers.items = localUsers;
+      this.localUsers.loading = false;
+    });
+    this.getGroupList().then((userGroups) => {
+      this.userGroups.items = userGroups;
+      this.userGroups.loading = false;
+    });
+  }
+
+  submitSearch() {
+    const identities = Object.keys(this.model.selectedIdentities).filter(
+      (urn) => !!this.model.selectedIdentities[urn],
+    );
+    const resources = Object.keys(this.model.selectedResources).filter(
+      (urn) => !!this.model.selectedResources[urn],
+    );
+    const actions = Object.keys(this.model.selectedActions).filter(
+      (action) => !!this.model.selectedActions[action],
+    );
+    const stateParams = {
+      identitiesFilter: identities,
+      resourcesFilter: resources,
+      actionsFilter: actions,
+    };
+    this.$state.go('^', stateParams);
```

Comment:
To defined in routing section ie
```suggestion
    this.goBack(stateParams);
```
In routing file, should be something like that 
```js
goBack: /* @ngInject */ ($state) => (params) => this.$state.go('^', params);
```